user.go

```diff
@@ -295,3 +295,21 @@ func (c *Client) PartialUpdateUsers(updates []PartialUserUpdate) (map[string]*Us
 
 	return resp.Users, err
 }
+
+type getOrCreateUserReq struct {
+	Users []userRequest `json:"users"`
+}
+
+// GetOrCreateUsers returns map of users; if user doesn't exist it will be created
+func (c *Client) GetOrCreateUsers(users []*User) (map[string]*User, error) {
+	var resp usersResponse
```

Comment:
this is my personal preference: 

I prefer variables next to where they are used. In this case, I prefer one line above `makeRequest`.